include/irr/asset/ICPUVirtualTexture.h

```diff
@@ -107,13 +106,13 @@ class ICPUVirtualTexture final : public IVirtualTexture<ICPUImageView>
             if (m_freePageTableLayerIDs.empty())
                 return STextureData::invalid();
             const uint32_t pgtLayer = m_freePageTableLayerIDs.top();
-            m_freePageTableLayerIDs.pop();
             pgtOffset = alloc(_img, _subres, pgtLayer);
             if ((pgtOffset==page_tab_offset_invalid()).all())//this would be super weird but let's check
                 return STextureData::invalid();
+            m_freePageTableLayerIDs.pop();
             storage->addPageTableLayer(pgtLayer);
 
-            (*m_layerToViewIndexMapping)[pgtLayer] = smplrIndex;
+            m_layerToViewIndexMapping[pgtLayer] = smplrIndex;
```

Comment:
weren't you supposed to map the LUT change flag here?

Reply:
what does it mean to map a flag?

Reply:
mark...

you update the LUT but never set the update flag... you have an unused update function

Reply:
lol XD yea, thx